test/datacollector.UnitTests/DataCollectionAttachmentManagerTests.cs

```diff
@@ -39,6 +44,47 @@ public void Cleanup()
         File.Delete(Path.Combine(TempDirectoryPath, "filename1.txt"));
     }
 
+    [TestMethod]
+    public void ParallelAccessShouldNotBreak()
+    {
+        string outputDirectory = Path.Combine(TempDirectoryPath, Guid.NewGuid().ToString());
```

Comment:
I don't know if the helper is available from this project but that might be better to use it if possible.

Reply:
It's not available...are unit tests and not integrations we're not inheriting from it unfortunately, maybe we should have something more "base" for everything.